### PR TITLE
Ft modify a request api 157733092

### PR DESCRIPTION
**Pull request by LuffyKing**
#### What does this PR do?
Adds a modifyARequest API for users registered on the app to modify their repair or maintenance requests,  and robust tests for said API.

#### Description of Task to be completed?
Addition of robust tests to ensure edge cases are accounted for and handled properly.  modifyARequest API created to allow users, who are registered on the app, to modify their requests
#### Any background context you want to provide?
Robust testing is the key to a solid application.
#### What are the relevant Github Issues?
The only issue so is only the pending review
Questions:

### CODE REVIEW

File: server/controllers/Requests.js [hunk 1]
```diff
@@ -121,27 +120,25 @@ const Requests = {
...
-        type: request.reqBody.type || result.type,
-      };
-      requests.push(updatedRequest);
-      return response.status(200).send({
-        message: 'Success - repair/maintenance request updated.',
-        updatedRequest
+    const { reqBody, decodedUser } = request;
+    const updateStatement = Object.keys(request.reqBody).map(key => `${key} = '${reqBody[key]}'`).join(',');
+    pool.connect((error, client, done) => {
+      if (error) response.status(500).send({ message: error.stack });
```
Comment (Tawakalt) [inline server/controllers/Requests.js:126]: Is it best practice to write `if statements` this way?
